src/plan_computation.py

```diff
@@ -110,4 +114,119 @@ def generate_photo_plan_on_grid(camera: Camera, dataset_spec: DatasetSpec) -> T.
+    """Compute the time of travel between two waypoints.
+    This function assumes the the drone is starting and ending with the same speed as in wp1.
+
+    Args:
+        wp1 (Waypoint): Waypoint of the starting position.
+        wp2 (Waypoint): Waypoint of the ending position.
+        max_speed (float): maximum speed (in meters per second)
+        max_acc (float): maximum acceleration (in meters per second squared)
+
+    Returns:
+        float: the minimum flight time (in seconds).
+
+    """
+
+    # Distance between waypoints.
+    d = math.sqrt((wp2.x - wp1.x)**2 + (wp2.y - wp1.y)**2)
+    v1 = wp1.speed
+
+    # d_acc is the distance needed to accelerate from initial speed to max_speed
+    # Calculated with basic kinematic equation d_acc = v1 * t + (1/2) * a * t**2
+    # with substitution of t using the relation of a * t = max_speed - v1.
+    # where:
+    #   d_acc = distance for accelerating from initial speed to max_speed
+    #   v1 = initial speed
+    #   a = acceleration
+    #   t = time for acceleration
+    d_acc = (v1 * (max_speed - v1) / max_acc
+                 + (1/2) * (max_speed - v1)**2 / max_acc)
+
+    # t_acc is the time needed for acceleration to max_speed
+    t_acc = (max_speed - v1) / max_acc
+
+    # If the drone is to attain max_speed and return back to initial speed,
+    # the distance between waypoints must be at least 2 * d_acc
+    # so there is enough distance to accelerate and decelerate.
+    # Any additional distance would be travelled at max_speed.
+    if d >= 2 * d_acc:
+        total_time = 2 * t_acc + (d - 2 * d_acc) / max_speed
+    else:
+        # In this case the highest speed during acceleratino
+        # can be calculated by solving the kinematic equation
+        # where we set the distance for acceleration to be d/2.
+        # d/2 = v1 * t + (1/2) a * t ** 2
+        # where:
+        #   t = (v2 - v1) / a
+        #   v2 = highest speed attained
+        #   v1 = initial speed
+        #   a = acceleration
+        v2 = math.sqrt(max_acc * d + v1 ** 2)
+        total_time = 2 * (v2 - v1) / max_acc
+    return total_time
+
+
+def total_time_for_flightplan(flightplan: T.List[Waypoint],
+                              max_speed: float,
+                              max_acc: float) -> float:
+    """Compute the time of travel for a flightplan (a list of Waypoints).
+
+    Args:
+        flightplan (T.List[Waypoint]): A flightplan represented as a list of Waypoints.
+        max_speed (float): The maximum drone speed.
+        max_acc (float): The maximum drone acceleration.
+
+    Returns:
+        float: the total flight time (in seconds).
+
+    """
+
+    total_time = 0.0
+    for i in range(1, len(flightplan)):
+        # Calculate and accumulate time between consecutive waypoints
+        time = time_between_waypoints(flightplan[i-1], flightplan[i],
+                                      max_speed, max_acc)
+        total_time += time
```

Comment:
Actually looping around here means you treated the vertical motion separately. Nice job!